### PR TITLE
Create model validation and test model specs

### PR DESCRIPTION
**Pull request by mohamed20163858**
in this sub-project I added unit tests for all models I created `User`, `Post`, `Comment`, and `Like`  using the RSpec-rails library also I added some validations like the name is not blank, post counter is greater or equal to zero and it must be an integer, Title must not be blank. Title must not exceed 250 characters, CommentsCounter must be an integer greater than or equal to zero, and LikesCounter must be an integer greater than or equal to zero. 
here is a screenshot after passing all tests
image

### CODE REVIEW

File: spec/model/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  # tests go here
+  subject { User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.') }
+  it 'name should be present' do
+    subject.name = nil
+    expect(subject).to_not be_valid
+  end
+  it 'PostsCounter must be an integer greater than or equal to zero' do
+    subject.posts_counter = nil
+    expect(subject).to_not be_valid
+  end
```
Comment (AnselemOdims) [inline spec/model/user_spec.rb]: - [ ] Good job working on these tests, but this does not validate that the `post_counter` is an integer. Please kindly include tests to actually validate that it is an integer and that it is not less than zero

File: spec/model/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    expect(post.likes_counter).to equal(0)
+  end
+  it 'CommentsCounter must be an integer greater than or equal to zero' do
+    post.comments_counter = nil
+    expect(post).to_not be_valid
+  end
+  it 'LikesCounter must be an integer greater than or equal to zero' do
+    post.likes_counter = nil
+    expect(post).to_not be_valid
+  end
```
Comment (AnselemOdims) [inline spec/model/post_spec.rb]: - [ ] Please kindly include tests to actually validate that CommentsCounter and LikesCounter are integers and that they are not less than zero

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @mohamed20163858 , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by omarsalem7 — APPROVED** [review]
Hi @mohamed20163858 😊, 
# STATUS: APPROVED! 🥇 💯 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
